packages/ui/src/components/Button/hooks/useItems.tsx

```diff
@@ -0,0 +1,91 @@
+import { ItemType as MenuItemType } from 'antd/lib/menu/hooks/useItems'
+import React, { useMemo } from 'react'
+
+import { sendAnalyticsClickEvent } from '../../_utils/analytics'
+import { Space } from '../../Space'
+import { Typography } from '../../Typography'
+import { DropdownButtonProps, ItemType } from '../dropdownButton'
+
+
+const DROPDOWN_CLASS_PREFIX = 'condo-dropdown'
+
+
+type UseItems = (items: Array<ItemType>, type: DropdownButtonProps['type']) => Array<MenuItemType>
+
+const convertItems = (items: Array<ItemType>, type: DropdownButtonProps['type']): Array<MenuItemType> => {
+    if (items.length < 1) return []
+
+    const resultItems: Array<MenuItemType> = []
+
+    for (let index = 0; index < items.length; index++) {
+        const item = items[index]
+
+        if (!item) continue
+
+        const { key, description, disabled, onClick, label, icon, id } = item
+        const mergedKey = key || index
+
+        const isItemWithIcon = item.icon
+        const isItemWithDescription = item.description
+
+        const withDividerAfterItem = index + 1 < items.length
+
+        const mergedItem: MenuItemType = {
+            key: mergedKey,
+            disabled,
+            onClick: (...args) => {
+                const stringContent = label
+
+                if (stringContent) {
+                    sendAnalyticsClickEvent('Button', { value: stringContent, type, id })
```

Comment:
Is it a button event? Sounds like dropdown one

Reply:
I also think that it should not include id of option, but id of button itself, because usually I want to receive a chosen option (value / label / key ?) and id of selector